simSPI/linear_simulator/params_utils.py

```diff
@@ -0,0 +1,361 @@
+"""contains functions and classes for parameter generation."""
+from abc import ABCMeta, abstractstaticmethod
+
+import numpy as np
+import starfile
+import torch
+import torch.fft
+from compSPI.distributions import uniform_to_triangular
+from ioSPI.starfile import check_star_file, starfile_opticsparams
+from pytorch3d.transforms import (
+    euler_angles_to_matrix,
+    matrix_to_euler_angles,
+    random_rotations,
+)
+
+
+def params_update(config):
+    """Update attributes of config.
+
+    Parameters
+    ----------
+    config: class
+        Class containing parameters of the dataset generator.
+
+    Returns
+    -------
+    config: class
+    """
+    config.starfile_available = config.input_starfile_path != ""
+    if config.starfile_available:
+        config = starfile_opticsparams(config)
+    config.ctf_size = config.side_len  # TODO: make it adaptable
```

Comment:
Make a github issue and take out the TODO
https://deepsource.io/gh/compSPI/simSPI/run/e173e44e-f4de-48fb-bedf-1e4c1fe17ad9/python/PYL-W0511

Reply:
Removed the TODO. It was some remnant from past.